Formula/licensed.rb

```diff
@@ -0,0 +1,33 @@
+class Licensed < Formula
+  desc "Cache and verify the licenses of dependencies"
+  homepage "https://github.com/github/licensed"
+  url "https://github.com/github/licensed.git",
+      tag:      "3.7.3",
+      revision: "76727f75d486a24758890a030e540ebf87bba78b"
+  license "MIT"
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "ruby@2.7"
+
+  def install
+    ENV["GEM_HOME"] = libexec
+    system "gem", "build", "licensed.gemspec"
+    system "gem", "install", "licensed-#{version}.gem"
+    bin.install libexec/"bin/licensed"
+    bin.env_script_all_files(libexec/"bin", GEM_HOME: ENV["GEM_HOME"])
+
+    # Avoid references to the Homebrew shims directory
+    shims_references = Dir[
+      libexec/"extensions/**/rugged-*/gem_make.out",
+      libexec/"extensions/**/rugged-*/mkmf.log",
+      libexec/"gems/rugged-*/vendor/libgit2/build/CMakeCache.txt",
+      libexec/"gems/rugged-*/vendor/libgit2/build/**/CMakeFiles/**/*",
+    ].select { |f| File.file? f }
+    inreplace shims_references, Superenv.shims_path.to_s, "<**Reference to the Homebrew shims directory**>", false
+  end
+
+  test do
+    assert_equal version.to_s, shell_output("#{bin}/licensed version").strip
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
@chenrui333 

Sorry, I forgot to replace that test.
I added the test with a mock project directory.

Reply:
👍